src/com/owncloud/android/files/services/FileDownloader.java

```diff
@@ -441,6 +452,39 @@ private void downloadFile(String downloadKey) {
                                     mCurrentDownload.getRemotePath()
                             );
 
+                    if (!downloadResult.isSuccess() && downloadResult.getException() != null) {
+
+                        // Check network availabality
+                        if (!ConnectivityUtils.isNetworkActive(getApplicationContext())) {
+
+                            // Schedule future retry of failed download due to network error from Android 5
+                            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+
+                                ComponentName mServiceComponent = new ComponentName(this,
+
+                                        RetryDownloadJobService.class);
+
+                                JobInfo.Builder builder;
+                                
+                                builder = new JobInfo.Builder(jobId, mServiceComponent);
+                                // require unmetered network
+                                builder.setRequiredNetworkType(JobInfo.NETWORK_TYPE_ANY);
```

Comment:
Comment says ```require unmetered network```, and so it should be; but parameter does not fit the comment; please, use ```JobInfo.NETWORK_TYPE_UNMETERED```.